src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.ui.Ui;
+import duke.util.TaskList;
+
+
+/**
+ * Duke is a personal assistant chatbot that helps a person to keep track of various things.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructs a Duke object.
+     */
+    public Duke() {
+        try {
+            this.ui = new Ui();
+            this.storage = new Storage();
+            this.tasks = new TaskList(storage.readTasks());
+            this.parser = new Parser();
+
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        }
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        try {
+            CommandType commandType = parser.parseCommandType(input);
+            return handleCommand(commandType, input);
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        }
+        return "Duke heard: " + input;
+    }
+
+    private void run() {
+        ui.printWelcomeMessage();
+        try (Scanner sc = new Scanner(System.in)) {
+            while (true) {
+                String input = sc.nextLine();
+                if (input.equals("bye")) {
+                    break;
+                }
+                CommandType commandType = parser.parseCommandType(input);
+                handleCommand(commandType, input);
+            }
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        } catch (Exception e) {
+            ui.printErrorMessage(new DukeException("An unexpected error occurred: " + e.getMessage()));
+        }
+        ui.printFarewellMessage();
+    }
+
+    /**
+     * Main entry point of the Duke application.
+     * @param args Command line arguments.
+     */
+    public static void main(String[] args) {
+        Duke ekud = new Duke();
+        ekud.run();
+    }
+
+    // private void handleCommand(CommandType commandType, String command) throws DukeException {
+    //     switch (commandType) {
+    //     case LIST:
+    //         ui.printList(tasks.getTasks());
+    //         break;
+    //     case MARK:
+    //         markTask(command);
+    //         break;
+    //     case DELETE:
+    //         deleteTask(command);
+    //         break;
+    //     case TODO:
+    //     case DEADLINE:
+    //     case EVENT:
+    //         addTask(command);
+    //         break;
+    //     case FIND:
+    //         handleFind(command);
+    //         break;
+    //     case UNKNOWN:
+    //         ui.printErrorMessage(new DukeException("I'm sorry, but I don't know what that means :-("));
+    //         break;
+    //     }
+    // }
```

Comment:
Mayhap can consider removing commented out code to improve readability.

Reply:
Right, thank you!